astropy/table/jsviewer.py

```diff
@@ -51,12 +51,30 @@ class Conf(_config.ConfigNamespace):
 """
 
 HTML_JS_SCRIPT = """
+var astropy_sort_num = function(a, b) {{
+    var a_num = parseFloat(a);
+    var b_num = parseFloat(b);
+
+    if (isNaN(a_num) && isNaN(b_num))
+        return ((a < b) ? -1 : ((a > b) ? 1 : 0));
+    else if (!isNaN(a_num) && !isNaN(b_num))
+        return ((a_num < b_num) ? -1 : ((a_num > b_num) ? 1 : 0));
+    else
+        return isNaN(a_num) ? -1 : 1;
+}}
+jQuery.extend( jQuery.fn.dataTableExt.oSort, {{
+    "optionalnum-asc": astropy_sort_num,
+    "optionalnum-desc": function (a,b) {{ return -astropy_sort_num(a, b); }}
+}});
+
 $(document).ready(function() {{
     $('#{tid}').dataTable({{
-        "order": [],
-        "iDisplayLength": {display_length},
-        "aLengthMenu": {display_length_menu},
-        "pagingType": "full_numbers"
+     "iDisplayLength": {display_length},
+     "aLengthMenu": {display_length_menu},
+     "pagingType": "full_numbers",
+     "bJQueryUI": true,
+     "sPaginationType": "full_numbers",
+     "aoColumnDefs": [{{"sType": "optionalnum", "aTargets": {sort_columns}}}]
```

Comment:
A few comments on these options:
- `order`: was added in  #4628 to avoid an initial sort of the table, I'm not sure if `sort_columns` gives the same behavior ?
- `bJQueryUI` : the style was changed to not use JQueryUI so this should be removed
- `sPaginationType` is the old name for `pagingType`
- `aoColumnDefs`: I think the new option name is [columnDefs](https://datatables.net/reference/option/columnDefs) but not sure (the API seems different: http://legacy.datatables.net/usage/columns)